### PR TITLE
Migration fixes

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**: This fixes the migration instructions.

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: -

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

### CODE REVIEW

File: migration/v0.7.x-v0.8.x/migrate-apps.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+   # Update everything else
+   ./bin/ck8s apply wc
+   ```
+9. Upgrade Service cluster applications
+   ```bash
+   # Delete old prometheus-operator release and related resources
+   ./bin/ck8s ops helm sc uninstall prometheus-operator -n monitoring
+   ./bin/ck8s ops kubectl sc delete service/prometheus-operator-kubelet -n kube-system
+   ./bin/ck8s ops kubectl sc delete pvc/prometheus-prometheus-operator-prometheus-db-prometheus-prometheus-operator-prometheus-0 -n monitoring
```
Comment (OlleLarsson) [inline migration/v0.7.x-v0.8.x/migrate-apps.md:67]: An awsomely long name for a pvc :rofl:

File: migration/v0.7.x-v0.8.x/migrate-apps.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+You may get warnings about missing values for some fluentd options in the Workload cluster.
+This can be disregarded.
+
+1. Checkout the new release: `git checkout v0.8.0`.
+2. Update helm to v3.4.1.
+3. The Helm repository `stable` has changed URL and has to be changed manually:
+   `helm repo add "stable" "https://charts.helm.sh/stable" --force-update`
+4. The blackbox chart has a changed dependency URL and has to be updated manually:
+   `cd helmfile/charts/blackbox && helm dependency update`
```
Comment (OlleLarsson) [inline migration/v0.7.x-v0.8.x/migrate-apps.md:14]:
```suggestion
   `cd helmfile/charts/blackbox && helm dependency update && cd ../../..`
```

**Review by OlleLarsson — APPROVED** [review]
Nice

File: migration/v0.7.x-v0.8.x/migrate-apps.md (hunk 1 above)
Comment (lentzi90) [reply migration/v0.7.x-v0.8.x/migrate-apps.md:67]: Indeed :rofl: no less than 5 `prometheus`!

Comment (OlleLarsson) [reply migration/v0.7.x-v0.8.x/migrate-apps.md:67]: Actually its **6** :laughing:

File: WIP-CHANGELOG.md [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
+    - `elasticsearch.tolerations` removed
+    - `elasticsearch.nodeSelector` removed
+    - `elasticsearch.affinity` removed
+    - `elasticsearch.storageClass` replaced by `elasticsearch.dataNode.storageClass`
+    - `elasticsearch.retention.kubeAuditSize` replaced by `elasticsearch.curator.retention.kubeAuditSizeGB`
+    - `elasticsearch.retention.kubeAuditAge` replaced by `elasticsearch.curator.retention.kubeAuditAgeDays`
+    - `elasticsearch.retention.kubernetesSize` replaced by `elasticsearch.curator.retention.kubernetesSizeGB`
+    - `elasticsearch.retention.kubernetesAge` replaced by `elasticsearch.curator.retention.kubernetesAgeDays`
+    - `elasticsearch.retention.otherSize` replaced by `elasticsearch.curator.retention.otherSizeGB`
+    - `elasticsearch.retention.otherAge` replaced by `elasticsearch.curator.retention.otherAgeDays`
```
Comment (viktor-f) [inline WIP-CHANGELOG.md:39]: Most if not all of these steps are included in the migration document. I think that's great but I think that also having it in the changelog is a bit redundant. I think you could change the whole `Release notes` to just include a reference to the migration document and a clear instruction that the operator must follow that document when upgrading.

**Review by viktor-f — COMMENTED** [review]
Very nice with one list of migration steps instead of several separated!

File: WIP-CHANGELOG.md (hunk 3 above)
Comment (lentzi90) [reply WIP-CHANGELOG.md:39]: Good point! I'll "minimize" the Release notes :+1:

**Pull request merged**